src/metabase/models/field_values.clj

```diff
@@ -125,6 +126,20 @@
                       {:type        type
                        :status-code 400})))))
 
+(defn- assert-no-identity-changes [field-values]
+  (let [{:keys [field_id] :as changes} (t2/changes field-values)]
+    (when (some #(contains? changes %) [:type :hash_key])
+      (throw (ex-info (tru "Can't update type or hash_key for a FieldValues.")
+                      {:id          (:id field-values)
+                       :type        (:type changes)
+                       :hash_key    (:hash_key changes)
+                       :status-code 400})))
+    (when (some? field_id)
+      (throw (ex-info (tru "Can't update field_id for a FieldValues.")
+                      {:id          (:id field-values)
+                       :field_id    field_id
+                       :status-code 400})))))
```

Comment:
this looks like it's almost the same thing in two ways, which seems arbitrary. why not combine them into one like this:
```suggestion
    (when (some #(contains? changes %) [:type :hash_key])
      (throw (ex-info (tru "Can't update type, hash_key, or field_id for a FieldValues.")
                      {:id          (:id field-values)
                       :type        (:type changes)
                       :hash_key    (:hash_key changes)
                       :field_id    (:field_id changes)
                       :status-code 400})))))
```